src/iris/pipelines/iris_pipeline.py

```diff
@@ -84,6 +84,19 @@ def __init__(
         self.nodes = self.instanciate_nodes()
         self.call_trace = self.env.call_trace_initialiser(nodes=self.nodes, pipeline_nodes=self.params.pipeline)
 
+    def update_config(self, config: str) -> None:
```

Comment:
I feel like some dedicated error raise might be needed to notify the user why the update fail (e.g. invalid ocnfig, invalid string ... )

Reply:
Yeah, good point. If you look closer, `_check_pipeline_coherency` method does that. I modified it slightly so that we don't have to override `self.params` (destroy the pipeline) before calling `_check_pipeline_coherency` method.